jung-algorithms/src/main/java/edu/uci/ics/jung/layout/model/LayoutModel.java

```diff
@@ -11,20 +11,40 @@
 
 import com.google.common.graph.Graph;
 import edu.uci.ics.jung.layout.algorithms.LayoutAlgorithm;
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
 import java.util.function.Function;
 
 /** two or three dimensional layoutmodel */
 public interface LayoutModel<N> extends Function<N, Point> {
 
+  /** @return the width of the layout area */
   int getWidth();
 
+  /** @return the height of the layout area */
   int getHeight();
 
+  /**
+   * allow the passed LayoutAlgorithm to operate on this LayoutModel
+   *
+   * @param layoutAlgorithm the algorithm to apply to this model's Points
+   */
   void accept(LayoutAlgorithm<N> layoutAlgorithm);
 
+  /**
+   * return a mapping of Nodes to Point locations
+   *
+   * @return
+   */
+  Map<N, Point> getLocations();
+
```

Comment:
Expose the method that is public in our only implementation. This allows the LayoutModel to deliver the single source of truth mapping between it's Graph Nodes and their locations.

Reply:
Just to make sure we're on the same page: the idea here is to deliver both the nodes (as the keys in the map) and the locations (values) in the same place, rather than needing to expose them separately?  seems reasonable.  :)